data/helpers.d/utils

```diff
@@ -370,72 +385,75 @@ ynh_get_plain_key() {
 ynh_read_manifest () {
-	# Declare an array to define the options of this helper.
-        local legacy_args=mk
-        declare -Ar args_array=( [m]=manifest= [k]=manifest_key= )
-        local manifest
-        local manifest_key
-	# Manage arguments with getopts
-	ynh_handle_getopts_args "$@"
-
-	if [ ! -e "$manifest" ]; then
-		# If the manifest isn't found, try the common place for backup and restore script.
-		manifest="../settings/manifest.json"
-	fi
-
-	jq ".$manifest_key" "$manifest" --raw-output
+    # Declare an array to define the options of this helper.
+    local legacy_args=mk
+    local -A args_array=( [m]=manifest= [k]=manifest_key= )
+    local manifest
+    local manifest_key
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+
+    if [ ! -e "$manifest" ]; then
+        # If the manifest isn't found, try the common place for backup and restore script.
+        manifest="../settings/manifest.json"
+    fi
+
+    jq ".$manifest_key" "$manifest" --raw-output
 }
 
 # Read the upstream version from the manifest
 #
+# usage: ynh_app_upstream_version [--manifest="manifest.json"]
+# | arg: -m, --manifest=    - Path of the manifest to read
+# | ret: the version number of the upstream app
+#
 # The version number in the manifest is defined by <upstreamversion>~ynh<packageversion>
 # For example : 4.3-2~ynh3
 # This include the number before ~ynh
 # In the last example it return 4.3-2
 #
-# usage: ynh_app_upstream_version [-m manifest]
-# | arg: -m, --manifest= - Path of the manifest to read
-#
 # Requires YunoHost version 3.5.0 or higher.
 ynh_app_upstream_version () {
     # Declare an array to define the options of this helper.
     local legacy_args=m
-    declare -Ar args_array=( [m]=manifest= )
+    local -A args_array=( [m]=manifest= )
     local manifest
     # Manage arguments with getopts
     ynh_handle_getopts_args "$@"
-
     manifest="${manifest:-../manifest.json}"
+
     version_key=$(ynh_read_manifest --manifest="$manifest" --manifest_key="version")
     echo "${version_key/~ynh*/}"
 }
 
 # Read package version from the manifest
 #
+# usage: ynh_app_package_version [--manifest="manifest.json"]
+# | arg: -m, --manifest=    - Path of the manifest to read
+# | ret: the version number of the package
+#
 # The version number in the manifest is defined by <upstreamversion>~ynh<packageversion>
 # For example : 4.3-2~ynh3
 # This include the number after ~ynh
 # In the last example it return 3
 #
-# usage: ynh_app_package_version [-m manifest]
-# | arg: -m, --manifest= - Path of the manifest to read
-#
 # Requires YunoHost version 3.5.0 or higher.
 ynh_app_package_version () {
     # Declare an array to define the options of this helper.
     local legacy_args=m
-    declare -Ar args_array=( [m]=manifest= )
+    local -A args_array=( [m]=manifest= )
     local manifest
     # Manage arguments with getopts
     ynh_handle_getopts_args "$@"
-
     manifest="${manifest:-../manifest.json}"
```

Comment:
> manifest can be a `.toml` file now!!

Sorry, I'm just trying to get all the places where we use "manifest.json"

Reply:
Could really be ??